pkg/hostagent/opflex.go

```diff
@@ -22,12 +22,59 @@ import (
 	"os"
 	"path/filepath"
 	"reflect"
+	"strings"
 	"text/template"
 
+	"encoding/json"
+	uuid "github.com/google/uuid"
 	"github.com/sirupsen/logrus"
 	"github.com/vishvananda/netlink"
 )
 
+type opflexFault struct {
+	FaultUUID   string `json:"fault_uuid"`
+	Severity    string `json:"severity"`
+	Description string `json:"description"`
+	FaultCode   int    `json:"faultCode"`
+}
+
+func writeFault(faultfile string, ep *opflexFault) (bool, error) {
+	newdata, err := json.MarshalIndent(ep, "", "  ")
+	if err != nil {
+		return true, err
+	}
+	existingdata, err := ioutil.ReadFile(faultfile)
+	if err == nil && reflect.DeepEqual(existingdata, newdata) {
+		return false, nil
+	}
+	err = ioutil.WriteFile(faultfile, newdata, 0644)
+	return true, err
+}
+
+func (agent *HostAgent) createFaultOnAgent(description string, faultCode int) {
+	Uuid := uuid.New().String()
+	FaultFilePath := filepath.Join(agent.config.OpFlexFaultDir, description+".fs")
```

Comment:
leading lower case is convention for local variables I think

Reply:
yes, will add this change